src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -88,10 +88,33 @@ class AppTasksResource @Inject() (service: MarathonSchedulerService,
                 @PathParam("taskId") id: String,
                 @QueryParam("scale") scale: Boolean = false): Response = {
     val pathId = appId.toRootPath
+
+    import mesosphere.util.ThreadPoolContext.context
+
     if (taskTracker.contains(pathId)) {
       val tasks = taskTracker.get(pathId)
       tasks.find(_.getId == id).fold(unknownTask(id)) { task =>
-        service.killTasks(pathId, Seq(task), scale)
+        if (scale) {
+          service.getApp(pathId) match {
```

Comment:
You probably copied this from other similar code but it is not ideal.

The `groupManager.updateApp` below passes the old app to the lambda function that you pass in.

The `updateApp` call itself is serialized and concurrent updates of the app definition are prevented. Your code contains a race condition. You update your app based on the result of the getApp call. If you have updates to the same app in-between, your code might discard them.